src/flows/pipes/Visualization/view.tsx

```diff
@@ -152,6 +162,164 @@ const Visualization: FC<PipeProp> = ({Context}) => {
+      }
+      if (threshold.min) {
+        fieldIndices[threshold.field].min = threshold.min
+      }
+      if (threshold.max) {
+        fieldIndices[threshold.field].max = threshold.max
+      }
+    })
+
+    const {columns} = results.parsed.table
+
+    let triggeredErrorThresholdMessage = ''
+
+    for (let i = 0; i < columns['_field'].data.length - 1; i++) {
+      const field = columns['_field'].data[i]
+      if (fieldIndices[`${field}`]) {
+        // find the value based on the indices
+        const value = columns['_value'].data[i]
+        const val = fieldIndices[`${field}`].value
+        switch (fieldIndices[`${field}`].type) {
+          case 'between':
+            if (
+              value > fieldIndices[`${field}`].min &&
+              value < fieldIndices[`${field}`].max
+            ) {
+              triggeredErrorThresholdMessage = `${field} is between ${
+                fieldIndices[`${field}`].min
+              } and ${fieldIndices[`${field}`].max}`
+            }
+            break
+          case 'not-between':
+            if (
+              value < fieldIndices[`${field}`].min &&
+              value > fieldIndices[`${field}`].max
+            ) {
+              triggeredErrorThresholdMessage = `${field} is not between ${
+                fieldIndices[`${field}`].min
+              } and ${fieldIndices[`${field}`].max}`
+            }
+            break
+          case 'not-equal':
+            if (value !== val) {
+              triggeredErrorThresholdMessage = `${field} is not equal to ${val}`
+            }
+            break
+          case 'equal':
+            if (value === val) {
+              triggeredErrorThresholdMessage = `${field} is equal to ${val}`
+            }
+            break
+          case 'less-equal':
+            if (value <= val) {
+              triggeredErrorThresholdMessage = `${field} is less than or equal to ${val}`
+            }
+            break
+          case 'less':
+            if (value < val) {
+              triggeredErrorThresholdMessage = `${field} is less than ${val}`
+            }
+            break
+          case 'greater-equal':
+            if (value >= val) {
+              triggeredErrorThresholdMessage = `${field} is greater than or equal to ${val}`
+            }
+            break
+          case 'greater':
+            if (value > val) {
+              triggeredErrorThresholdMessage = `${field} is greater than ${val}`
+            }
+            break
+          default:
+            break
+        }
+      }
+
+      if (triggeredErrorThresholdMessage) {
+        break
+      }
+    }
+
+    if (triggeredErrorThresholdMessage) {
+      // show the QR code if it exists
+      const url = new URL(
+        `${window.location.origin}${window.location.pathname}?panel=${id}`
+      ).toString()
+      const doc = document.getElementById(
+        `${triggeredErrorThresholdMessage}-${id}`
+      )
+      const size = doc?.offsetHeight * 0.4 ?? 100
```

Comment:
what determines these consts?

Reply:
This is for the size of the QR code. Totally arbitrary with regards to the size, just using a height relative to the offset of the pipe to generate the size of the QR code. Figured design might have input when we come to that point